spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -0,0 +1,418 @@
+      end
+
+    end
+
+    context 'when an invalid response code is returned' do
+
+      it 'reports the error and returns nil' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_INVALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        expect(subject.abs_get_resource_hosts).to eq(test_expected_result)
+      end
+
+    end
+
+    context 'when an invalid response body is returned' do
+
+      it 'reports the error and returns nil' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(test_http_response)
+
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        expect(subject.abs_get_resource_hosts).to eq(test_expected_result)
+      end
+
+    end
+
+    context 'when the response is nil' do
+
+      it 'reports the error and returns nil' do
+        ENV['ABS_TOKEN'] = TEST_ABS_TOKEN
+        test_expected_result = nil
+
+        expect(subject).to receive(:abs_request_awsdirect).at_least(:once).and_return(nil)
+        expect(subject.abs_get_resource_hosts).to eq(test_expected_result)
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_last_abs_resource_hosts' do
+
+    context 'when the file exists' do
+
+      context 'when the file contains a valid list of hosts' do
+
+        it 'returns the hosts' do
+          skip
+        end
+
+      end
+
+      context 'when the file does not contain a valid list of hosts' do
+
+        it 'reports the error and returns nil' do
+          skip
+        end
+
+      end
+
+      context 'when the file is empty' do
+
+        it 'reports the error and returns nil' do
+          skip
+        end
+
+      end
+
+    end
+
+    context 'when the file does not exist' do
+
+      it 'reports the error and returns nil' do
+        skip
+      end
+
+    end
+
+  end
+
+  describe '#abs_return_resource_hosts' do
+
+    test_uri = TEST_AWSDIRECTRETURN_URI
+    test_body = TEST_AWSDIRECTRETURN_REQUEST_BODY
+
+    context 'when an array of hosts is specified via ABS_RESOURCE_HOSTS' do
+
+      it 'submits requests to return the hosts and returns the host array' do
+        ENV['ABS_RESOURCE_HOSTS'] = TEST_ABS_RESOURCE_HOSTS
+
+        expect(subject).to receive(:abs_get_base_url).at_least(:once).and_return(TEST_BASE_URL)
```

Comment:
might want to spec that this should be called twice, to make sure it does both hosts

Reply:
This should work for any number of hosts.